solc/CommandLineParser.cpp

```diff
@@ -0,0 +1,1243 @@
+		signatureHashes == _other.signatureHashes &&
+		natspecDev == _other.natspecDev &&
+		natspecUser == _other.natspecUser &&
+		ast == _other.ast;
+}
+
+bool CommandLineOptions::operator==(CommandLineOptions const& _other) const
+{
+	return
+		sourceFilePaths == _other.sourceFilePaths &&
+		standardJsonInputFile == _other.standardJsonInputFile &&
+		remappings == _other.remappings &&
+		addStdin == _other.addStdin &&
+		basePath == _other.basePath &&
+		allowedDirectories == _other.allowedDirectories &&
+		ignoreMissingInputFiles == _other.ignoreMissingInputFiles &&
+		errorRecovery == _other.errorRecovery &&
+		outputDir == _other.outputDir &&
+		overwriteFiles == _other.overwriteFiles &&
+		evmVersion == _other.evmVersion &&
+		experimentalViaIR == _other.experimentalViaIR &&
+		revertStrings == _other.revertStrings &&
+		stopAfter == _other.stopAfter &&
+		inputMode == _other.inputMode &&
+		targetMachine == _other.targetMachine &&
+		inputAssemblyLanguage == _other.inputAssemblyLanguage &&
+		libraries == _other.libraries &&
+		prettyJson == _other.prettyJson &&
+		coloredOutput == _other.coloredOutput &&
+		withErrorIds == _other.withErrorIds &&
+		selectedOutputs == _other.selectedOutputs &&
+		estimateGas == _other.estimateGas &&
+		combinedJsonRequests == _other.combinedJsonRequests &&
+		metadataHash == _other.metadataHash &&
+		metadataLiteral == _other.metadataLiteral &&
+		optimize == _other.optimize &&
+		expectedExecutionsPerDeployment == _other.expectedExecutionsPerDeployment &&
+		noOptimizeYul == _other.noOptimizeYul &&
+		yulOptimiserSteps == _other.yulOptimiserSteps &&
+		initializeModelChecker == _other.initializeModelChecker &&
+		modelCheckerSettings == _other.modelCheckerSettings;
+}
+
+bool CommandLineParser::parseInputPathsAndRemappings()
+{
+	m_options.ignoreMissingInputFiles = (m_args.count(g_argIgnoreMissingFiles) > 0);
+	if (m_args.count(g_argInputFile))
+		for (string path: m_args[g_argInputFile].as<vector<string>>())
+		{
+			auto eq = find(path.begin(), path.end(), '=');
+			if (eq != path.end())
+			{
+				if (auto r = ImportRemapper::parseRemapping(path))
+					m_options.remappings.emplace_back(std::move(*r));
+				else
+				{
+					serr() << "Invalid remapping: \"" << path << "\"." << endl;
+					return false;
+				}
+
+				string remappingTarget(eq + 1, path.end());
+				m_options.allowedDirectories.insert(boost::filesystem::path(remappingTarget).remove_filename());
+			}
+			else if (path == "-")
+				m_options.addStdin = true;
+			else
+				m_options.sourceFilePaths.insert(path);
+		}
+
+	return true;
+}
+
+bool CommandLineParser::parseLibraryOption(string const& _input)
+{
+	namespace fs = boost::filesystem;
+	string data = _input;
+	try
+	{
+		if (fs::is_regular_file(_input))
+			data = readFileAsString(_input);
+	}
+	catch (fs::filesystem_error const&)
+	{
+		// Thrown e.g. if path is too long.
+	}
+	catch (FileNotFound const&)
+	{
+		// Should not happen if `fs::is_regular_file` is correct.
+	}
```

Comment:
For the `filesystem_error` case we could generate a human readable message so the user knows why it failed. If the catch below should not happen, we could make it `solAssert(false, "reason here");`?

Reply:
Agreed. Again, something I did not want to touch yet in this PR.